### ISSUE TITLE
Support building the Ingress Container with Podman

### ISSUE THREAD

**Issue by aknot242** [nginxinc/kubernetes-ingress#2924]
**Is your feature request related to a problem? Please describe.**
The current image build is Docker-centric, and you cannot build the ingress controller with Podman.

**Describe the solution you'd like**
The instructions, Makefile and Dockerfile should support building the ingress container with podman.

**Describe alternatives you've considered**
Attempted to simply `alias` podman to docker, however this alone does not work since unix alias [will not expand into non-interactive shells](https://askubuntu.com/questions/98782/how-to-run-an-alias-in-a-shell-script). Additionally, a number of changes need to be made to the Makefile and Dockerfile to make them compatible with the feature set that podman supports (such as removing `--link` flags).

**Additional context**

**Comment by aknot242** [comment]
Correct, @lucacome. The request is about **building** with podman ([technically, buildah](https://podman.io/blogs/2018/10/31/podman-buildah-relationship.html), to your point). Use case summary is being able to build it locally (on a developer's desktop) using freely available tools in light of Docker Desktop's licensing changes.

**Comment by brianehlert** [comment]
docker as a cli and build tool is not tied to Docker Desktop nor docker desktop licensing.
You can simply install the docker cli without Docker Desktop and be able to build.  I don't believe anything changed there.

**Comment by codygreen** [comment]
This is only true for Linux machines.  There is no docker-cli for Mac or Windows and the docker FAQ states the user must install a linux virtual machine.

> If you use a Mac, you can install Docker CLI and Engine inside a virtual machine, using VirtualBox or VMware Fusion for example, which may require purchasing a license for VirtualBox or VMware Fusion. On Windows you could install and run Docker CLI and Engine inside WSL2.
> If you use a Linux machine you can easily use the Docker CLI and Docker Engine, See [the documentation on installing Docker Engine](https://docs.docker.com/engine/install/#server) for instructions. Docker Desktop is not yet available for Linux.
https://www.docker.com/pricing/faq/

Ideally, we want users to be able to build the NIC containers natively on any platform they want.

**Comment by brianehlert** [comment]
Sorry, I have been a Windows user and Admin for over 20 years.
if you need to build images on Windows use WSL2 and install Docker.  It isn't a VM.
https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-containers

All that Docker Desktop does is bring hooks into WSL and set up VM hooks and ease of use into docker running in a VM.
Running containers is a different debate.

This is about the investment of an entirely different image building framework that only recently was even capable of doing what we require.

**Comment by codygreen** [comment]
I think we're getting off track here... this request is about removing operational complexity and making it easier for more people to build and use NIC.

**Comment by lucacome** [comment]
@codygreen @aknot242 I understand (and share) the desire to make it easier for people to use the Ingress Controller. That's why we have pre-built images for as many OSs and architectures as possible. Is there something missing that makes you want to build the images manually? Maybe that's the real issue.

I think the normal (and recommended) use case for the majority of people is to use one of the images that we publish (and please feel free to open an issue if you think something is missing) since they have our "stamp of approval" 🙂  but even if somebody wants to build the image manually, Docker is still the most popular by far and removing features just to make it compatible with buildah doesn't seem like a really good trade-off, at least for now. 

This is only the second request (?) that we received about buildah (https://github.com/nginxinc/kubernetes-ingress/issues/2490) since the change in Docker Desktop licensing that was mentioned, which went into effect in August 2021, so that makes me think that the alternatives to Docker are not very popular, but I'm happy to be proven wrong.

TL;DR my suggestion is to use pre-built images as much as possible and open issues/proposals about expanding on that.

**Comment by lucacome** [comment]
Linking https://github.com/containers/buildah/pull/4295 that will reduce the number of warnings and https://github.com/containers/podman/issues/15423 that is a blocker at the moment.

**Comment by brianehlert** [comment]
Still not hearing the feedback on this that we would like to prioritize it at this time.
Closing.  Will revisit if it gains additional traction through Discussions, Ideas
